### PR TITLE
Fixed argparse help summary when no docstring was provided

### LINKED ISSUE #561: Should help verbose use description from ArgumentParser?
The `help -v` details for the built-in commands are all short descriptions, however the details for argparse-wrapped commands describe the syntax of the commands. This seems sort of inconsistent for a default, would make more sense for it to prefer the ArgumentParser description first? I mean, I could copy the descriptions to the docstrings if I wanted to just fix this myself, but I am wondering if a change in the way cmd2 does it is more appropriate.

Example "weird-looking" result of this:

```
Documented commands (type help <topic>):
================================================================================
alias               Define or display aliases
catann              usage: catann [-h] [-c COURSE] [ids [ids ...]]
cc                  usage: cc [-h] [course]
cd                  usage: cd [-h] [directory]
edit                Edit a file in a text editor.
help                List available commands with "help" or detailed help with "help cmd".
history             View, run, edit, save, or clear previously entered commands.
la                  usage: la [-h] [-c COURSE] [-l] [-s] [-u]
lann                usage: lann [-h] [-c COURSE] [-n NUMBER] [-d DAYS]
lc                  usage: lc [-h] [-a] [-l] [-i]
lg                  usage: lg [-h] [-c COURSE] [-l] [-u] [-a]
load                Runs commands in script file that is encoded as either ASCII or UTF-8 text.
login               usage: login [-h] [-q] url token
pullf               usage: pullf [-h] [-c COURSE] [-o OUTPUT]
py                  Invoke python command, shell, or script
pyscript            Runs a python script file inside the console
quit                Exits this application.
set                 Sets a settable parameter or shows current settings of parameters.
shell               Execute a command as if at the OS prompt.
shortcuts           Lists shortcuts (aliases) available.
unalias             Unsets aliases
whoami              usage: whoami [-h] [-v]
wopen               usage: wopen [-h] [-c COURSE] [tabs [tabs ...]]
```

### PR DESCRIPTION
**Pull request by kmvanbrunt**
Fixes #561

### CODE REVIEW

**Comment by marklalor** [comment]
I think this may have somehow uncategorized any argparser commands that are missing an argparse description (the "undocumented" ones still have the `"Clanvas"` category set by `@cmd2.with_category`

```
Documented commands (type help <topic>):

Clanvas
================================================================================
catann              Print course announcements.
la                  List course assignments.
lann                List course announcements.
lc                  List courses.
lg                  List course grades.
login               Set URL and token to use for all Canvas API calls
pullf               Pull course files to local disk.
ua                  Upload a submission to an assignment
wopen               Open tabs in canvas web interface.

Other
================================================================================
alias               Manage aliases
cd                  Change the working directory.
edit                Edit a file in a text editor
help                List available commands or provide detailed help for a specific command
history             View, run, edit, save, or clear previously entered commands
load                Run commands in script file that is encoded as either ASCII or UTF-8 text
macro               Manage macros
py                  Invoke Python command or shell
pyscript            Run a Python script file inside the console
quit                Exit this application
set                 Set a settable parameter or show current settings of parameters
shell               Execute a command as if at the OS prompt
shortcuts           List available shortcuts

Undocumented commands:
======================
cc  whoami
```

File: cmd2/cmd2.py [hunk 1]
```diff
@@ -211,16 +213,14 @@ def cmd_wrapper(instance, cmdline):
 
         # argparser defaults the program name to sys.argv[0]
         # we want it to be the name of our command
-        argparser.prog = func.__name__[3:]
+        argparser.prog = func.__name__[len(COMMAND_FUNC_PREFIX):]
```
Comment (marklalor) [inline cmd2/cmd2.py]: Nice change removing those magic numbers.

**Comment by kmvanbrunt** [comment]
@marklalor I'll take a look at the category thing.

File: cmd2/cmd2.py [hunk 2]
```diff
@@ -211,16 +213,14 @@ def cmd_wrapper(instance, cmdline):
...
         # If the description has not been set, then use the method docstring if one exists
         if argparser.description is None and func.__doc__:
             argparser.description = func.__doc__
 
-        if func.__doc__:
-            setattr(cmd_wrapper, HELP_SUMMARY, func.__doc__)
-
-        cmd_wrapper.__doc__ = argparser.format_help()
+        # If argparser.description is None, this will show under 'Undocumented commands'
+        cmd_wrapper.__doc__ = argparser.description
```
Comment (marklalor) [inline cmd2/cmd2.py]: Could possibly just default this to the previous `argparser.format_help()` value?

Comment (kmvanbrunt) [reply cmd2/cmd2.py]: That would put the usage statement back, which I think looks odd in a help summary.

**Comment by kmvanbrunt** [comment]
@marklalor Try the category thing now. I made it so a command with a category will always display under that category's section whether or not it has documentation.

**Comment by marklalor** [comment]
@kmvanbrunt works as you described. I like this solution because it's consistent all around and this "awkward gap" can be fixed by simply going in and actually adding some descriptions!

```
Documented commands (type help <topic>):

Clanvas
================================================================================
catann              Print course announcements.
cc
la                  List course assignments.
lann                List course announcements.
lc                  List courses.
lg                  List course grades.
login               Set URL and token to use for all Canvas API calls
pullf               Pull course files to local disk.
ua                  Upload a submission to an assignment
whoami
wopen               Open tabs in canvas web interface.

Other
================================================================================
alias               Manage aliases
cd                  Change the working directory.
edit                Edit a file in a text editor
help                List available commands or provide detailed help for a specific command
history             View, run, edit, save, or clear previously entered commands
load                Run commands in script file that is encoded as either ASCII or UTF-8 text
macro               Manage macros
py                  Invoke Python command or shell
pyscript            Run a Python script file inside the console
quit                Exit this application
set                 Set a settable parameter or show current settings of parameters
shell               Execute a command as if at the OS prompt
shortcuts           List available shortcuts
```

**Comment by kmvanbrunt** [comment]
@marklalor I agree that it's more consistent now. A category is enough information to keep a command from being labeled undocumented.

File: cmd2/cmd2.py [hunk 3]
```diff
@@ -120,14 +120,16 @@ def __subclasshook__(cls, C):
...
 INTERNAL_COMMAND_EPILOG = ("Notes:\n"
                            "  This command is for internal use and is not intended to be called from the\n"
                            "  command line.")
 
 # All command functions start with this
-COMMAND_PREFIX = 'do_'
+COMMAND_FUNC_PREFIX = 'do_'
+
+# All help functions start with this
+HELP_FUNC_PREFIX = 'help_'
```
Comment (tleonhardt) [inline cmd2/cmd2.py]: Good call breaking this out as a constant too.

**Pull request merged**